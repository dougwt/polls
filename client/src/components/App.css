body {
  background-color: #fcfcfc;
}
a {
  color: #06bad6;
}
a:visited:hover,
a:hover {
  color: #ef476f;
}
a:visited {
  color: #7c265d;
}

.btn:hover,
.btn-large:hover {
  background-color: #06d6a0;
  color: #fcfcfc;
}

.btn:hover {
}

.teal {
  background-color: #06d6a0 !important;
}

.red {
  background-color: #ef476f !important;
}
