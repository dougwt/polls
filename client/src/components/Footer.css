footer,
.page-footer {
  background-color: #06bad6;
  line-height: 1.8rem;
}

.page-footer a:hover {
  color: #ef476f !important;
}

.page-footer .nav ul {
  font-family: 'Raleway', cursive;
  font-size: 0.9rem;
}

.footer-copyright a {
  color: #ffd166;
  font-weight: bold;
}

.footer-copyright .heart {
  color: #ef476f;
}
