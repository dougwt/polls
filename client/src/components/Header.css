nav {
  background-color: #06bad6;
}
nav ul a,
nav ul a:visited,
nav ul a:hover,
nav ul a:visited:hover {
  margin-bottom: 2px;
  color: #fcfcfc;
}
nav ul a:hover,
nav ul a:visited:hover {
  background-color: #ef476f;
}
nav ul {
  font-family: 'Raleway', cursive;
}
nav ul .toggle {
  display: none;
}

.nav-wrapper.container {
  margin-top: 0;
}
.brand-logo img {
  max-height: 40px;
  margin: 10px 0;
}

@media screen and (max-width: 992px) {
  nav {
    z-index: 9;
  }

  nav ul .responsive {
    display: none;
  }

  nav.unfold ul .responsive {
    display: block;
    width: 100%;
    background-color: #06bad6;
    position: absolute;
    top: 64px;
    right: 0;
    z-index: 10;
  }
  nav.unfold ul .responsive li {
    display: block;
    width: 100%;
  }
  nav.unfold ul .responsive a {
    display: block;
  }

  nav ul .toggle {
    display: block;
    z-index: 1000;
  }
}

@media screen and (max-width: 600px) {
  nav.unfold ul .responsive {
    top: 56px;
  }
}
