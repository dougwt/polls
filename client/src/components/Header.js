import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link, NavLink } from 'react-router-dom';

const renderContent = auth => {
  switch (auth) {
  case null:
    return;
  case false:
    return (
      <li>
        <a
          href="/auth/google"
          className="navbar-item waves-effect waves-light"
        >
          <i className="material-icons left">person</i>
            Sign In with Google
        </a>
      </li>
    );
  default:
    return (
      <li>
        <a
          href="/api/logout"
          className="navbar-item waves-effect waves-light"
        >
          <i className="material-icons left">person</i>
            Sign Out
        </a>
      </li>
    );
  }
};

export const Header = props => {
  return (
    <nav>
      <div className="nav-wrapper container">
        <Link to="/" className="brand-logo">
          Simple Polls
        </Link>

        <ul className="right">
          <li>
            <NavLink to="/polls" className="waves-effect waves-light">
              <i className="material-icons left">dashboard</i>
              Dashboard
            </NavLink>
          </li>
          <li>
            <NavLink to="/polls/new" className="waves-effect waves-light">
              <i className="material-icons left">add</i>
              Create a Poll
            </NavLink>
          </li>
          {renderContent(props.auth)}
        </ul>
      </div>
    </nav>
  );
};
Header.propTypes = {
  auth: PropTypes.object
};

function mapStateToProps(state) {
  return { auth: state.auth };
}

export default connect(mapStateToProps)(Header);
