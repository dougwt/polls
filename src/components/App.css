@import url('https://fonts.googleapis.com/css?family=Raleway');
body {
  background-color: #fcfcfc;
}
a {
  color: #06bad6;
}
a:visited:hover,
a:hover {
  color: #ef476f;
}
a:visited {
  color: #7c265d;
}

.btn:hover,
.btn-large:hover {
  background-color: #06d6a0;
  color: #fcfcfc;
}

.btn:hover {
}

.teal {
  background-color: #06d6a0 !important;
}

.red {
  background-color: #ef476f !important;
}

h1,
h2,
h3,
h3,
h4,
h5 {
  font-family: 'Raleway', cursive;
}

h3 {
  margin-bottom: 1.2em;
}
