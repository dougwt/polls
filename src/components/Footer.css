footer,
.page-footer {
  background-color: #06bad6;
  line-height: 1.8rem;
}

.page-footer p {
  font-size: 0.9rem;
}

.page-footer a:hover {
  color: #ef476f !important;
}

.page-footer .nav ul {
  font-family: 'Raleway', cursive;
  font-size: 0.9rem;
}

.page-footer a,
.page-footer a:visited {
  color: #f9cd69;
}

.footer-copyright a {
  color: #ffd166;
  font-weight: bold;
}

.footer-copyright .heart {
  color: #ef476f;
}
