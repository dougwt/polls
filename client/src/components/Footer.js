import React from 'react';
import './Footer.css';

const Footer = () => {
  return (
    <footer className="page-footer">
      <div className="container">
        <div className="row">
          <div className="col l9 s12">
            <h5 className="white-text">Simple Polls</h5>
            <p className="grey-text text-lighten-4">
              Lorem ipsum dolor sit amet, consectetur adipisicing elit.
              Dignissimos earum veritatis numquam beatae harum fuga culpa itaque
              quaerat minus omnis accusantium ipsam aperiam esse mollitia,
              tempore porro, a quasi libero!
            </p>
          </div>
          <div className="col l3 s12 nav">
            <h5 className="white-text">Site Navigation</h5>
            <ul>
              <li>
                <a className="white-text" href="#!">
                  Homepage
                </a>
              </li>
              <li>
                <a className="white-text" href="#!">
                  Dashboard
                </a>
              </li>
              <li>
                <a className="white-text" href="#!">
                  Create a Poll
                </a>
              </li>
              <li>
                <a className="white-text" href="#!">
                  GitHub Repo
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>

      <div className="footer-copyright">
        <div className="container">
          Made with <span className="heart">&#9829;</span> by{' '}
          <a className="text-lighten-3" href="https://dougwt.com">
            @dougwt
          </a>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
