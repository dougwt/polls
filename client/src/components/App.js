import React, { Component } from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../actions';
import Header from './Header';
import Footer from './Footer';
import Landing from './Landing';
import Dashboard from './Dashboard';
import PollNew from './polls/PollNew';
import PollDetail from './polls/PollDetail';

export class App extends Component {
  componentDidMount() {
    this.props.fetchUser();
  }

  render() {
    return (
      <BrowserRouter>
        <div className="app">

          <Header />

          <Route exact path="/" component={Landing} />
          <Route exact path="/polls" component={Dashboard} />
          <Route exact path="/polls/new" component={PollNew} />
          <Route exact path="/polls/1" component={PollDetail} />

          <Footer />

        </div>
      </BrowserRouter>
    );
  }
}

export default connect(null, actions)(App);
