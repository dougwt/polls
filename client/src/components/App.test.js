import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';
import Header from './Header';
import Footer from './Footer';

describe('App', () => {
  let props = {};

  let app = shallow(<App {...props} />);

  it('renders properly', () => {
    expect(app).toMatchSnapshot();
  });

  it('shows a Header', () => {
    expect(app.find(Header).exists()).toBe(true);
  });

  it('shows a Footer', () => {
    expect(app.find(Footer).exists()).toBe(true);
  });
});
