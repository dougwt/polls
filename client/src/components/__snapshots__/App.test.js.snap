// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="app"
>
        <Connect(Header) />
        <Route
                component={[Function]}
                exact={true}
                path="/"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/polls"
        />
        <Route
                component={Object {}}
                exact={true}
                path="/polls/new"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/polls/1"
        />
        <Footer />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/polls"
/>,
          <Route
            component={Object {}}
            exact={true}
            path="/polls/new"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/polls/1"
/>,
          <Footer />,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/polls",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": Object {},
            "exact": true,
            "path": "/polls/new",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/polls/1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="app"
>
          <Connect(Header) />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/polls"
          />
          <Route
                    component={Object {}}
                    exact={true}
                    path="/polls/new"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/polls/1"
          />
          <Footer />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Header) />,
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/polls"
/>,
            <Route
              component={Object {}}
              exact={true}
              path="/polls/new"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/polls/1"
/>,
            <Footer />,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/polls",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": Object {},
              "exact": true,
              "path": "/polls/new",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/polls/1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
